indico/util/fs.py

```diff
@@ -144,17 +144,41 @@ def cleanup_dir(path, min_age, dry_run=False, exclude=None):
         has_files = False
         for filename in files:
             filepath = os.path.join(root, filename)
-            if os.path.getmtime(filepath) >= min_mtime:
+
+            try:
+                # raises an exception for invalid symlinks
+                mtime = os.path.getmtime(filepath)
+            except OSError:
```

Comment:
Is it always a `FileNotFoundError` in case of broken symlinks? If yes, we could catch that one separately and in case of other OSErrors just keep the logging and `has_files = True`

Reply:
If it's a symlink, then I think it's safe to assume that `ENOENT` means a broken symlink. But I think `ENOENT` can be raised by files which are not symlinks, so I don't think that the implication `ENOENT` => broken symlink is valid. Conversely, a broken symlink can also show up as other errors - `ELOOP` (which is detected by `is_broken_symlink`) and `ENOTDIR` which right now is ignored.

Reply:
Based on the MM conversation, I keep it as it is now ;)